docs/api/rest.md

```diff
@@ -3,6 +3,10 @@ title: "REST API"
 ---
 import ApiEndpoint from '@site/static/js/api_endpoint.jsx'
 
+:::warning
+The REST API has been superseded by the WebSocket API and will no longer be updated.
```

Comment:
The 2024.4 update moved the weather forecast from an attribute that could be fetched via the REST API to a service call whose return data cannot be read by the REST API.

It seems that I’m not alone in viewing that as _losing a feature_, especially since a fix was proposed (and implemented!) but project maintainers turned it away because “the REST API has been superseded by the WebSocket API”.

I expect more data to be moved to service call responses over time as that feature is relatively new, making the REST API less useful over time. If Home Assistant wants it to still be a viable feature (which seems to be the desire of the community), a new endpoint needs to be added to support service response data. Or a parameter to the existing service endpoint, but I’m not reopening that discussion here.

Unless the project is willing to allow that change, calling it deprecated doesn’t seem inaccurate to me.

Reply:
The attributes on the weather entries were deprecated and removed, not the rest API that serves entity attributes.

Reply:
Yes, but how do I access the weather  forecast with the REST API now?

Reply:
@allenporter, @emontnemery's [answer on the architecture discussion](https://github.com/home-assistant/architecture/discussions/1074#discussioncomment-9196867) says (emphasis mine)

> In the cases where the same functionality is offered by both the REST and websocket APIs, as is the case with service calls, **we should deprecate the REST endpoint** instead of adding new features to it.

Reply:
That response says that specific endpoints will be deprecated when they are deprecated. It doesn't say that all are actively being turned off.  Does that make sense?

Reply:
> It doesn't say that all are actively being turned off.

Neither does this PR.

We could update this PR to explicitly mark endpoints that currently have a WebSocket counterpart (i.e. everything except for `history/period`, `logbook`, `camera_proxy`, `calendars`, `template`, and `intent/handle`) with a symbol that indicates that those will no longer be updated, and a warning that no new ones will be added.

Reply:
I may have misread the original writing in my response so I'll retract what I said above. I think what I'm reacting to is the chatter that folks think they are forced to now use unsuitable websocket APIs,
and that existing endpoints are going away across the board or something.

I can see this PR doesn't exactly say that.

Reply:
I mean, they *are* forced to do that: you can no longer access forecast data without using the websocket API, and the stated direction doesn't inspire much confidence in the long term survival of the REST API.